examples/next/pages/ui/components/storage/storage-manager/metadata/index.page.tsx

```diff
@@ -0,0 +1,40 @@
+import { Amplify } from 'aws-amplify';
+import { withAuthenticator } from '@aws-amplify/ui-react';
+import {
+  StorageManager,
+  StorageManagerProps,
+} from '@aws-amplify/ui-react-storage';
+import '@aws-amplify/ui-react/styles.css';
+import awsExports from './aws-exports';
+Amplify.configure(awsExports);
+
+const processFile: StorageManagerProps['processFile'] = ({
+  file,
+  key,
+  metadata,
+}) => {
+  return {
+    file,
+    key,
+    metadata: {
+      ...metadata,
+      id: key,
+    },
+  };
+};
+
+export function StorageManagerExample() {
+  return (
+    <StorageManager
+      acceptedFileTypes={['image/*']}
+      accessLevel="private"
+      maxFileCount={3}
+      showThumbnails={true}
+      metadata={{
+        test: 'bar',
+      }}
+      processFile={processFile}
+    />
+  );
+}
```

Comment:
Based on this example, it's unclear to me why we need to add the `metadata` prop, since a user could just do this?
```tsx
const metadata = { test: 'bar' }

const processFile: StorageManagerProps['processFile'] = ({
  file,
  key,
}) => {
  return {
    file,
    key,
    metadata: {
      ...metadata,
      id: key,
    },
  };
};

export function StorageManagerExample() {
  return (
    <StorageManager
      acceptedFileTypes={['image/*']}
      accessLevel="private"
      maxFileCount={3}
      showThumbnails={true}
      processFile={processFile}
    />
  );
}
```

Reply:
Yea this example is kinda bad. I was adding `metadata` on the StorageManager so users could add non-file-specific metadata without having to use `processFile`. If you are already using `processFile` there would be no need to use both at the same time.

Reply:
IMO we should really just offer one prop since having two ways to do this can cause confusion and potentially lead the user to use both erroneously. 

Since we already support `processFile` what do you think of only updating the `ProcessFile` interface to return `metadata`?